crates/starknet-types-core/benches/criterion_hashes.rs

```diff
@@ -0,0 +1,108 @@
+use criterion::{BatchSize, BenchmarkId, black_box, Criterion, criterion_group, criterion_main};
+use rand::{RngCore, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+use starknet_types_core::felt::Felt;
+use starknet_types_core::hash::{Pedersen, Poseidon};
+use starknet_types_core::hash::StarkHash;
+
+const SEED: u64 = 3;
+
+fn rnd_felt(rng: &mut ChaCha8Rng) -> Felt {
+    let mut felt: [u8; 32] = Default::default();
+    rng.fill_bytes(&mut felt);
+    Felt::from_bytes_be(&felt)
+}
+
+fn rnd_felts(rng: &mut ChaCha8Rng, n: usize) -> Vec<Felt> {
+    (0..n).map(|_| rnd_felt(rng)).collect()
+}
+
+fn rnd_pair_of_felts(rng: &mut ChaCha8Rng) -> (Felt, Felt) {
+    (rnd_felt(rng), rnd_felt(rng))
+}
+
+fn poseidon_hash_single_input(c: &mut Criterion) {
+    let mut rng = ChaCha8Rng::seed_from_u64(SEED);
+    let mut group = c.benchmark_group("Poseidon");
+
+    let (x, y) = rnd_pair_of_felts(&mut rng);
+
+    group.bench_function("hash - single input", |bench| {
+        bench.iter(|| black_box(Poseidon::hash(&x, &y)))
+    });
+}
+
+fn poseidon_hash_multiple_inputs(c: &mut Criterion) {
+    let mut rng = ChaCha8Rng::seed_from_u64(SEED);
+    let mut group = c.benchmark_group("Poseidon");
+
+    group.bench_function("hash - multiple inputs", |bench| {
+        bench.iter_batched(
+            || rnd_pair_of_felts(&mut rng),
+            |(x, y)| black_box(Poseidon::hash(&x, &y)),
+            BatchSize::SmallInput
+        );
+    });
+}
+
+fn poseidon_hash_array(c: &mut Criterion) {
+    let mut rng = ChaCha8Rng::seed_from_u64(SEED);
+
+    let mut group = c.benchmark_group("Poseidon");
+
+    for n in [2, 4, 8, 16].iter() {
```

Comment:
I think we need to test with larger input sizes
```suggestion
    for n in [10, 100, 1000, 10000].iter() {
```

Reply:
Out of curiosity, what is the use case for hashing 10000 long vec of felts?

Reply:
I think it makes sense since calculating the Sierra class hash for a given Sierra file involves doing `Poseidon(SierraProgramFelts)`. Depending on the contract it can be a huge vector of felts.

Reply:
can be up to 81k for class hash
Also the transaction hash requires a hash chain for the calldata which can be up to 4k
https://docs.starknet.io/documentation/tools/limits_and_triggers/
so def add the 10k plz

Reply:
done